keras_nlp/layers/cached_multi_head_cross_attention.py

```diff
@@ -0,0 +1,98 @@
+# Copyright 2023 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Cached multi-head cross-attention layer based on `keras.layers.MultiHeadAttention`."""
+
+import tensorflow as tf
+from tensorflow import keras
+
+from keras_nlp.api_export import keras_nlp_export
+
+
+@keras_nlp_export("keras_nlp.layers.CachedMultiHeadCrossAttention")
+class CachedMultiHeadCrossAttention(keras.layers.MultiHeadAttention):
```

Comment:
what is the main difference between this and `CachedMultiHeadAttention`? Ideally the other layer could work in either a cross attn of self attention context.